### PR TITLE
GHA/windows: formatting, adjust timeouts, tidy-ups

### PR DESCRIPTION
**Pull request by vszakats**
- reduce job timeouts for msys2 and old-mingw-w64 jobs.
- fold long lines and reorder options in MSVC jobs.
- make some options default in MSVC jobs.

### CODE REVIEW

**Comment by dfandrich** [comment]
Analysis of PR #15017 at [52695294](https://github.com/curl/curl/commit/52695294fc8403df8d29cc6c4b3d302fe0438c32):

[Test 445 failed](https://github.com/curl/curl/actions/runs/10994260164/job/30522772930#check-step-15), which has NOT been flaky recently, so **there could be a real issue in the PR**.

###### Generated by [Testclutch](https://testclutch.curl.se/)

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -525,14 +549,22 @@ jobs:
             #          read its configuration from, making it vulnerable to attacks on
             #          Windows. Do not use this component till there is a fix for these.
             # https://github.com/curl/curl-for-win/blob/3951808deb04df9489ee17430f236ed54436f81a/libssh.sh#L6-L8
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=OFF -DCURL_USE_LIBSSH=ON -DCURL_USE_GNUTLS=ON -DCURL_USE_GSASL=ON -DUSE_NGTCP2=ON'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:552]:
```suggestion
            config: >
```

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -479,42 +479,66 @@ jobs:
...
+              -DCURL_CA_SEARCH_SAFE=ON
+
           - name: 'boringssl-ECH'
             install: 'brotli zlib zstd libpsl nghttp2 boringssl libssh2[core,zlib]'
             arch: 'x64'
             plat: 'windows'
             type: 'Debug'
             tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_HTTPSRR=ON -DUSE_ECH=ON'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:526]:
```suggestion
            config: >
```

File: .github/workflows/windows.yml [hunk 3]
```diff
@@ -479,42 +479,66 @@ jobs:
             plat: 'windows'
             type: 'Debug'
             tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=ON  -DCURL_USE_SCHANNEL=ON  -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DCURL_USE_MBEDTLS=ON -DCURL_USE_GNUTLS=ON -DCURL_DEFAULT_SSL_BACKEND=schannel -DCURL_USE_GSASL=ON -DUSE_WIN32_IDN=ON'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:482]:
```suggestion
            config: >
```

File: .github/workflows/windows.yml [hunk 4]
```diff
@@ -479,42 +479,66 @@ jobs:
...
+              -DCURL_USE_GSASL=ON -DUSE_WIN32_IDN=ON -DENABLE_UNICODE=ON
+
           - name: 'openssl'
             install: 'brotli zlib zstd libpsl nghttp2 nghttp3 openssl libssh2 pkgconf gsasl c-ares libuv'
             arch: 'x64'
             plat: 'windows'
             type: 'Debug'
             tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON -DCURL_USE_GSASL=ON -DENABLE_ARES=ON -DCURL_USE_LIBUV=ON'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:493]:
```suggestion
            config: >
```

File: .github/workflows/windows.yml [hunk 5]
```diff
@@ -479,42 +479,66 @@ jobs:
...
+              -DCURL_USE_GSASL=ON -DENABLE_ARES=ON -DCURL_USE_LIBUV=ON
+
           - name: 'openssl'
             install: 'brotli zlib zstd        nghttp2 nghttp3 openssl libssh2'
             arch: 'x64'
             plat: 'uwp'
             type: 'Debug'
             tflags: 'skiprun'
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON -DCURL_USE_LIBPSL=OFF'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:504]:
```suggestion
            config: >
```

File: .github/workflows/windows.yml [hunk 6]
```diff
@@ -479,42 +479,66 @@ jobs:
...
+              -DUSE_HTTPSRR=ON -DUSE_ECH=ON
+
           - name: 'wolfssl'
             install: 'brotli zlib zstd libpsl nghttp2 wolfssl libssh2 pkgconf gsasl ngtcp2[wolfssl] nghttp3'
             arch: 'x64'
             plat: 'windows'
             type: 'Debug'
             tflags: '~1516'
-            config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_BROTLI=ON -DCURL_ZSTD=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_WOLFSSL=ON -DUSE_NGTCP2=ON -DCURL_USE_GSASL=ON'
+            config: |
```
Comment (vszakats) [inline .github/workflows/windows.yml:537]:
```suggestion
            config: >
```

**Pull request closed without merging**